silk/tools/otns_manager.py

```diff
@@ -531,3 +639,67 @@ def unsubscribe_from_all_nodes(self):
     for subscriber in self.otns_monitor_map.values():
       subscriber.unsubscribe()
     self.otns_monitor_map.clear()
+
+  def update_layout(self):
+    """Update layout of nodes in auto layout mode.
+    """
+    if not self.auto_layout or not self.otns_node_map:
+      return
+
+    self.logger.debug("Updating nodes layout")
+    first_node = next(iter(self.otns_node_map))
+    center_x, center_y = first_node.device.get_otns_vis_layout_center()
+    radius = first_node.device.get_otns_vis_layout_radius()
+
+    routers = set()
+    children = set()
+    detached = set()
+    extaddr_to_node_map = {}
+
+    for node in self.otns_node_map.values():
+      extaddr_to_node_map[node.extaddr] = node
+      if node.role == RoleType.DISABLED or node.role == RoleType.DETACHED:
+        detached.add(node)
+      elif node.role == RoleType.CHILD:
+        children.add(node)
+      elif node.role == RoleType.ROUTER or node.role == RoleType.LEADER:
+        routers.add(node)
+
+    if not routers and not children and not detached:
+      return
+
+    # order children near their parents
+    router_list = list(routers)
+    router_list.sort(key=lambda x: x.node_id)
+
+    children_list = []
+    for router in router_list:
+      for child in router.children:
+        if child in extaddr_to_node_map:
+          children_list.append(extaddr_to_node_map[child])
+          children.discard(extaddr_to_node_map[child])
+    children_list.sort(key=lambda x: x.node_id)
+
+    non_attached_children_list = list(children)
+    non_attached_children_list.sort(key=lambda x: x.node_id)
+    children_list.extend(non_attached_children_list)
+
+    detached_list = list(detached)
+    detached_list.sort(key=lambda x: x.node_id)
+
+    groups = [g for g in [router_list, children_list, detached_list] if g]
+    groups_count = len(groups)
+    # adding a shift to prevent overlapping
+    shift = 15
+    for i, group in enumerate(groups):
+      n = len(group)
+      if n == 1 and i == 0:
+        group[0].update_vis_position(center_x, center_y)
+      else:
+        angle_step = math.radians(360 / n)
+        group_radius = radius * (i + 1) / groups_count
+        for j, node in enumerate(group):
+          angle = shift * i + angle_step * j
```

Comment:
do we put children near their parent?

Reply:
At least order them near their parents using the code above.